Metalama.Framework.Engine/Templating/TemplateAnnotator.cs

```diff
@@ -2297,6 +2292,19 @@ private void ReportUnsupportedLanguageFeature( SyntaxNodeOrToken nodeForDiagnost
 
     public override SyntaxNode VisitLocalFunctionStatement( LocalFunctionStatementSyntax node )
     {
+        foreach ( var attributeList in node.AttributeLists )
+        {
+            foreach ( var attribute in attributeList.Attributes )
+            {
+                var attributeTypeSymbol = (this._syntaxTreeAnnotationMap.GetSymbol( attribute.Name ) as IMethodSymbol)?.ContainingType;
+
+                if ( attributeTypeSymbol != null && attributeTypeSymbol.AllInterfaces.Any( i => i.Name == nameof(ITemplateAttribute) ) )
```

Comment:
I think this can be extended to any attribute derived from `ScopeAttribute` because local functions are now run-time-only by default and compile-time ones are not supported.